extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/AnnotatedUnremovableClassesProcessor.java

```diff
@@ -0,0 +1,39 @@
+package io.quarkus.arc.deployment;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.AnnotationTarget;
+import org.jboss.jandex.DotName;
+
+import io.quarkus.arc.Unremovable;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+
+public class AnnotatedUnremovableClassesProcessor {
+
+    private static final DotName UNREMOVABLE = DotName.createSimple(Unremovable.class.getName());
+
+    @BuildStep
+    public void findAnnotatedClasses(BeanArchiveIndexBuildItem beanArchiveIndexBuildItem,
+            BuildProducer<UnremovableBeanBuildItem> producer) {
+        final Set<AnnotationTarget> namesOfClassesMarkedAsUnremovable = new HashSet<>();
```

Comment:
Well, it's not `namesOfClassesMarkedAsUnremovable` anymore. Maybe `unremovableBeanTargets` or something like that?

Reply:
Yes of course! I had set my mind to change it and then I forgot.... Sorry about this dump omission

Reply:
Fixed